lxd/cluster/gateway.go

```diff
@@ -958,19 +946,29 @@ func (g *Gateway) currentRaftNodes() ([]db.RaftNode, error) {
 	// Get the names of the raft nodes from the global database.
 	if g.Cluster != nil {
 		err = g.Cluster.Transaction(func(tx *db.ClusterTx) error {
+			nodes, err := tx.GetNodes()
+			if err != nil {
+				return fmt.Errorf("Failed loading cluster members: %w", err)
+			}
+
+			nodesByAddress := make(map[string]db.NodeInfo, len(nodes))
+			for _, node := range nodes {
+				nodesByAddress[node.Address] = node
+			}
+
 			for i, server := range servers {
-				node, err := tx.GetNodeByAddress(server.Address)
-				if err != nil {
-					return err
+				node, found := nodesByAddress[server.Address]
+				if !found {
+					return fmt.Errorf("Cluster member info not found for %q", server.Address)
```

Comment:
Nice, definitely better than constant queries. Using a map for the `server.Address` lookup is pretty clever too :)

Reply:
Yeah saves repeatedly iterating the same slice for mapping :)